src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1009,7 +1009,7 @@ private function addInlineReference(string $id, Definition $definition, string $
         $name = $this->getNextVariableName();
         $this->referenceVariables[$targetId] = new Variable($name);
 
-        $reference = ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE >= $behavior ? new Reference($targetId, $behavior) : null;
+        $reference = ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE !== $behavior ? new Reference($targetId, $behavior) : null;
```

Comment:
Perhaps, there was just a typo here and the author meant `$behavior >= ContainerInterface::EXCEPTION_ON_INVALID_REFERENCE`. 

The bug was hidden for a long time because it can only be reproduced for "synthetic" services like `kernel` and in specific cases. But the other services replaced by null on optimization step before